docs/adr/0003-graphql-layer-between-backend-frontend.md

```diff
@@ -0,0 +1,98 @@
+# GraphQL layer between backend and frontend
+
+Date: 2020-04-30
+
+## Status
+
+**Accepted**
+
+## Related tickets (not available for public)
+
+SIG-2568
+
+## Context
+
+Since our current backend only serves data through REST and there is no possibility to generate
+types and or interfaces, there is no also no automated way to validate data on the frontend
+which is coming from the backend.
+
+Besides of that there are too many inconstencies between different backend calls and the current
+backend API documentation is not always correct and/or up to date because it's written manually
+and not generated from code.
```

Comment:
Django Rest Framework offers the ability to automatically generate an OpenAPI specification from the code with `python manage.py generateschema`. Maybe we can use that?

Reply:
Great info, thanks!

I would love to have access to those schema's since it could help to make the thin `layer` described in this `ADR` better integrated with the backend.

Reply:
Think I can find some time next week to create a pull request for this.